tests/flows/erc20_source_erc20_destination.go

```diff
@@ -163,3 +163,158 @@ func ERC20SourceERC20Destination(network interfaces.Network) {
 	Expect(err).Should(BeNil())
 	Expect(balance).Should(Equal(bridgedAmount))
 }
+
+/**
+ * Deploy an ERC20 token source on the primary network
+ * Defers deploying ERC20Destination to Subnet A
+ * Bridges C-Chain example ERC20 tokens to Subnet A
+ * Confirms failure to execute message, since destination hasn't been deployed.
+ * Deploys ERC20Destination to Subnet A
+ * Invokes a retry of message execution
+ * Confirms message execution
+ */
+func ERC20DestinationNotYetDeployed(network interfaces.Network) {
+	cChainInfo := network.GetPrimaryNetworkInfo()
+	subnetAInfo, _ := teleporterUtils.GetTwoSubnets(network)
+	fundedAddress, fundedKey := network.GetFundedAccountInfo()
+
+	ctx := context.Background()
+
+	// Deploy an ExampleERC20 on the primary network as the source token to be bridged
+	sourceTokenAddress, sourceToken := teleporterUtils.DeployExampleERC20(
+		ctx,
+		fundedKey,
+		cChainInfo,
+	)
+
+	// Create an ERC20Source for bridging the source token
+	erc20SourceAddress, erc20Source := utils.DeployERC20Source(
+		ctx,
+		fundedKey,
+		cChainInfo,
+		fundedAddress,
+		sourceTokenAddress,
+	)
+
+	// Token representation on subnet A will have same name, symbol, and decimals
+	tokenName, err := sourceToken.Name(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+	tokenSymbol, err := sourceToken.Symbol(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+	tokenDecimals, err := sourceToken.Decimals(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+
+	// pre-determine the deployment address of the ERC20Destination, but
+	// don't deploy it yet
+	erc20DestinationAddressNonce, err := subnetAInfo.RPCClient.NonceAt(
+		ctx,
+		fundedAddress,
+		nil,
+	)
+	Expect(err).Should(BeNil())
+	erc20DestinationAddressNonce += 1
+	erc20DestinationAddress := crypto.CreateAddress(
+		fundedAddress,
+		erc20DestinationAddressNonce,
+	)
+	erc20Destination, err := erc20destination.NewERC20Destination(
```

Comment:
Nit, it looks like `erc20Destination` isn't used until around line 308. Could we move it's create to just prior to its use? 

(At first I was confused why were creating it before trying to bridge the tokens, but then realized it isn't related to the actual deployment of the contract. Either way, think it will be most clear if it's only created when needed)

Reply:
agree, lets move the nonce and expected address calculation below too if not used until then

Reply:
the nonce and expected address calculation are needed at this point because the address of the not-yet-deployed `ERC20Destination` needs to be included in the `SendTokensInput` struct that's sent for the execution that's expected to fail.